### ISSUE TITLE
Can't install with Firefox 60

### ISSUE THREAD

**Issue by gabri25ele** [Noitidart/Chrome-Store-Foxified#117]
After Firefox update to latest (60) version, the Chrome Store Foxified don't install anything.
It stay ever on "Checking AMO Credentials" and don't go on.

**Comment by lipatovroman** [comment]
Same issue. 

Endless validating when using link from Chrome store (See attachments)

![2018-05-11_13-02-04](https://user-images.githubusercontent.com/6747127/39919271-bf6396b4-551b-11e8-8405-26f04ad35484.jpg)

Preparing to download and install when installing extension from https://chrome.google.com/webstore

![2018-05-11 13-04-41](https://user-images.githubusercontent.com/6747127/39919363-09844c16-551c-11e8-83e9-e9426700c217.jpg)

And finally, I have downloaded CRX extension https://biblsoft.ru/windows/internet/ad-blockers/168-adblock-plus and tried to install it - no results:

![2018-05-11_13-08-29](https://user-images.githubusercontent.com/6747127/39919489-7745fa4c-551c-11e8-9dac-059c89210923.jpg)

**Comment by Two-Tone** [comment]
Same here.  Trying to install the Google Dictionary add-on, but it's just been endlessly checking

**Comment by Two-Tone** [comment]
I've emailed the dev to ask him about this extension and bug.  He had been far too busy recently to look at CSF but is now looking into this bug as it's not even working for him when he tested it.

**Comment by Noitidart** [comment]
Thanks very much @Two-Tone for reaching out. I'm taking a look right now and finding a fetch is now getting blocked for "Cross-Origin Request". This didn't happen before. Am looking into it.

**Comment by Noitidart** [comment]
Thanks @znmeb yep it is. I hope to knock both of these bugs out. For any extension devs out there, it seems the problem is I am not able to do a `fetch` request to `https://addons.mozilla.org/` anymore. And this is where the API for signing lives. I hope this doesn't mean Foxified is no longer possible as an extension.

**Comment by QWERTYUIOPYOZO** [comment]
hmm...I will take a look at this.

**Comment by Noitidart** [comment]
Thanks @QWERTYUIOPYOZO - I actually just got confirmation from a Mozilla engineer that they have blocked `fetch` to `https://addons.mozilla.org/`. Even the `https://addons.mozilla.org/api/` end points do not have CORs headers. We would have to ask the AMO engineers to add CORS headers.

And for the `fetch` to `https://addons.mozilla.org/` I'll have to ask people to manually input the user token and password as I can't inject content scripts into this.

The new extension system of Firefox seems to be turning into more trash day by day. It's real generic of me to say this, but it just feels like "block fetch to addons.mozilla.org" is not a real solution to whatever issue they were trying to fix.

**Comment by Two-Tone** [comment]
Did they explain why they blocked it?

**Comment by Noitidart** [comment]
Nope no explanation. I asked for a bug maybe they didn't see my comment yet. Hopefully they will link me.

**Comment by QWERTYUIOPYOZO** [comment]
Well that's annoying.

**Comment by Noitidart** [comment]
Got explanation "No. We don't block fetch, we just don't grant special permissions for AMO based on add-on permissions anymore." So this means the cross origin special permission doesn't apply to AMO. So fetch isn't "blocked" but it has been "disabled".

**Comment by QWERTYUIOPYOZO** [comment]
Come on Mozilla. Same difference.

**Comment by QWERTYUIOPYOZO** [comment]
You run a server to host the compatibility forum.
Is there a reason you don't just have the extension send the URL of the chrome extension to the server where it can be temporarily downloaded to the server, then uploaded to AMO and signed through a single purpose-created account (which you would know the API key for), and then stored on the server for the client to download. If it works, it would eliminate the need for the user to have an AMO account,  upload/download the extension multiple times and convert extensions that have already been converted by other users.

**Comment by Two-Tone** [comment]
That could get expensive and it's a bit much to expect him to do it for
free.

**Comment by QWERTYUIOPYOZO** [comment]
Hmm. It would be much easier if Mozilla simply integrated this straight into firefox.

**Comment by Noitidart** [comment]
I used to use a single-purpose account, but then the addon reviewers rejected it and required me to use user accounts. Maybe I can ask them, due to the changes if I can go back to single purpose account and hard code in account token for signing.

That server way is for sure the way to go. I already have a server for the forum. Just need to set it up to do all the signing now too.

Yes totally. It would be very much easier if Mozilla just let me do it in the extension.

**Comment by bluenote73** [comment]
It's not clear to me from reading this, is it possible to work around this temporarily by downgrading my firefox?

**Comment by Noitidart** [comment]
@bluenote73 it probably will work. I'm not sure in what version they disabled fetch to AMO.

I am going to try to set up my server to do this stuff now.

**Comment by CRC-Mismatch** [comment]
@bluenote73 @znmeb Installing 59.0.3 to a different dir just as a workaround seems to do it for now. The hassle is that you have to make sure that install won't update, and will sync CSF/extensions with your current one.
I think it could work via Firefox Sync with different profiles, but just to check, I was able to make it work just setting Firefox to never update, close the current one, launch the old one, redo the CSF plugin install, close the old one and reopen the current one.

**Comment by QWERTYUIOPYOZO** [comment]
This problem seems quite ironic, considering webextensions' goal is to prevent addons from breaking when Firefox is updated.

**Comment by znmeb** [comment]
@CRC-Mismatch I only had one extension that needed this tool, and they finally made an XPI that will install on Firefox 60. So I don't need a workaround any more.

**Comment by ShintoPlasm** [comment]
Agreed with **QWERTYUIOPYOZO**: it's ridiculous that one of the key 'benefits' of WEs seems to have been undermined by Mozilla.

**Comment by chaoscreater** [comment]
Hi guys, see my workaround highlighted here:
https://github.com/Noitidart/Chrome-Store-Foxified/issues/113

**Comment by bluenote73** [comment]
Any new news on this item, maybe from Firefox?

**Comment by bluenote73** [comment]
For anyone suffering with this, I worked around by switching to Yandex mobile browser on Android which supports native chrome extensions.
